code/modules/arousal/arousal.dm

```diff
@@ -145,7 +145,7 @@
 			partners -= L
 		if(iscarbon(L))
 			var/mob/living/carbon/C = L
-			if(!C.exposed_genitals.len && !C.is_groin_exposed() && !C.is_chest_exposed()) //Nothing through_clothing, no proper partner.
+			if(!C.exposed_genitals.len || !C.is_groin_exposed() || !C.is_chest_exposed() || !C.is_mouth_covered()) //Nothing through_clothing, no proper partner.
```

Comment:
This is an inverse conditional check checking for invalid conditions and removing viable elements from the list, which means you need to be checking for your condition not applying. Do this instead:
```suggestion
			if(!C.exposed_genitals.len && !C.is_groin_exposed() && !C.is_chest_exposed() && C.is_mouth_covered()) //Nothing through_clothing, no proper partner.
```

Reply:
this would require them to all be uncovered instead of just one which was talked about in the discord server. also your proposed change says C.is_mouth_covered() instead of !C.is_mouth_covered()

Reply:
^ what lin said

Reply:
Look at the next line of code. This is a check filtering invalid elements which can be seen by the "candidates -=" in the next line, as opposed to a "+="
The current "&&" implementation is already an or, as !(A && B) is the same as (!A || !B).
Any of the things need to be false for it to not remove the element, aka, at least one of the conditions needs to be true

Reply:
This is also the reason why it checks for if it IS covered, because that being the case *as well as* all other things being covered it should remove it from the candidates.

Reply:
I see that now, but wouldn't C.is_mouth_covered() still need the negation?

Reply:
Ah ok my b